### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by ssaru**
1. train, test 모드를 required으로 변경했습니다.
2. 기본 default batch size를 16으로 변경했습니다.
3. 학습 초기에만 step / loss 해상도를 높이고, 학습이 길어질 수록 해상도를 낮추는 방식으로 변경했습니다.
( visdom의 read/write load를 감소시키면서, loss의 경향은 조금 더 잘 볼 수 있게끔)
4. visdom에서 epoch와 step을 결합해서 total step을 계산하는 수식이 잘못되어 수정했습니다.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+from test import test
+
+parser = argparse.ArgumentParser(description='YOLO v1.')
+parser.add_argument('--mode',                      type=str,   help='train or test', default='train')
+parser.add_argument('--dataset',                   type=str,   help='dataset to train on, voc', default='voc')
+parser.add_argument('--data_path',                 type=str,   help='path to the data', required=True)
+parser.add_argument('--class_path',                type=str,   help='path to the filenames text file', required=True)
+parser.add_argument('--input_height',              type=int,   help='input height', default=448)
+parser.add_argument('--input_width',               type=int,   help='input width', default=448)
+parser.add_argument('--batch_size',                type=int,   help='batch size', default=16)
```
[the remark this answers is not in the document]
Comment (ssaru) [reply main.py]: 2의 n승!!

해당도를 낮춘다는 말은, 매 step마다 loss를 기록하는게  train할 때, load가 드는 일이니까 학습 초반에 loss 기록을 짧은 주기 t에서 기록하다가 나중에 train step이 어느정도 지나면 loss 기록을 긴 주기 hat_t에서 한다는 의미입니다.

File: train.py [hunk 2]
```diff
@@ -137,13 +137,18 @@ def train(params):
 
     # Train the model
     total_step = len(train_loader)
+
+    total_train_step = num_epochs * total_step
```
[the remark this answers is not in the document]
Comment (ssaru) [reply train.py]: epoch는 전체 step에 대해서 얼마나 iteration을 도느냐
total_step은 전체 데이터를 batch size로 나누었을 때의 개수입니다.
따라서 전체 얼마나 train step을 밟았는지 확인하려면 epoch * total_step이 되야지 알 수 있다고 생각했습니다.
변수명이 헷갈리면 변경하도록 하겠습니다.
